### PR TITLE
[NodeSearchBundle] Bugfix for indexing StructuredNodes and children

### PR DESCRIPTION
**Pull request by treeleaf**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Fixed tickets | #1509, #1547, #1606

### CODE REVIEW

File: src/Kunstmaan/NodeSearchBundle/Configuration/NodePagesConfiguration.php [hunk 1]
```diff
@@ -245,12 +246,15 @@ public function indexNodeTranslation(NodeTranslation $nodeTranslation, $add = fa
             return false;
         }
 
+        // Retrieve the referenced entity from the public NodeVersion
+        $page = $publicNodeVersion->getRef($this->em);
+
         $node = $nodeTranslation->getNode();
         if ($this->isIndexable($page)) {
```
Comment (sandergo90) [inline src/Kunstmaan/NodeSearchBundle/Configuration/NodePagesConfiguration.php]: @treeleaf Maybe use the ReflectionClass here to ? And only when it's indexable you retrieve the ref ?

Comment (treeleaf) [inline src/Kunstmaan/NodeSearchBundle/Configuration/NodePagesConfiguration.php]: Ok, I did oversee that part. I have modified it.

File: src/Kunstmaan/NodeSearchBundle/Configuration/NodePagesConfiguration.php [hunk 2]
```diff
@@ -683,4 +687,21 @@ protected function getAclPermissions($object)
...
+     * @param $publicNodeVersion
+     *
+     * @return mixed
+     */
+    private function getNodeRefPage(NodeVersion $publicNodeVersion)
+    {
+        $refEntityName = $publicNodeVersion->getRefEntityName();
+
+        if (!isset($this->nodeRefs[$refEntityName])) {
+            $refEntityClass = new \ReflectionClass($refEntityName);
```
Comment (Devolicious) [inline src/Kunstmaan/NodeSearchBundle/Configuration/NodePagesConfiguration.php]: Reflection class seems to be obsolete here. Just `new $refEntityName()` should do the trick in this case.

File: src/Kunstmaan/NodeSearchBundle/EventListener/NodeIndexUpdateEventListener.php [hunk 3]
```diff
@@ -44,7 +44,9 @@ public function onPostPersist(NodeEvent $event)
     private function index(NodeEvent $event)
     {
         $nodeSearchConfiguration = $this->container->get('kunstmaan_node_search.search_configuration.node');
-        $nodeSearchConfiguration->indexNodeTranslation($event->getNodeTranslation(), true);
+        $nodeTranslation = $event->getNodeTranslation();
+        $nodeSearchConfiguration->indexNodeTranslation($nodeTranslation, true);
+        $nodeSearchConfiguration->indexChildren($event->getNode(), $nodeTranslation->getLang());
```
Comment (mlebkowski) [inline src/Kunstmaan/NodeSearchBundle/EventListener/NodeIndexUpdateEventListener.php]: what is the purpose of this addition?

Comment (treeleaf) [inline src/Kunstmaan/NodeSearchBundle/EventListener/NodeIndexUpdateEventListener.php]: This is to address issue #1547; to reindex children when the parent url changes for instance.

Comment (mlebkowski) [inline src/Kunstmaan/NodeSearchBundle/EventListener/NodeIndexUpdateEventListener.php]: Wouldn’t this reindex the whole site if you save the home page?

Comment (dannyvw) [inline src/Kunstmaan/NodeSearchBundle/EventListener/NodeIndexUpdateEventListener.php]: Yes that's right

Comment (mlebkowski) [inline src/Kunstmaan/NodeSearchBundle/EventListener/NodeIndexUpdateEventListener.php]: Are we sure we want to do this? It takes over a dozen of seconds to do this on my production machine, not to mention on the dev server and unnecessary load on elasticsearch, etc.

Comment (Devolicious) [inline src/Kunstmaan/NodeSearchBundle/EventListener/NodeIndexUpdateEventListener.php]: I think that we can solve this with a quick check to see if the URL has changed and if it has, than reindex it's children.

Comment (mlebkowski) [inline src/Kunstmaan/NodeSearchBundle/EventListener/NodeIndexUpdateEventListener.php]: Can we please think about abstracting it into some kind of `ChildrenNeedReindexingDecisionMakerInterface`? I have nodes that index elements of their parents, so this kind of behaviour would apply there.

Comment (mlebkowski) [inline src/Kunstmaan/NodeSearchBundle/EventListener/NodeIndexUpdateEventListener.php]: Disregard my last. It’s on an event either way, so any additional logic can be attached in client bundles.

**Pull request merged**